### PR TITLE
Support Oculus SDK version 0.6 on windows only

### PR DESCRIPTION
**Pull request by jherico**
Oculus' 0.6 SDK greatly improves the OpenGL compatibility and resolves some issues, but it has a drastically different API from 0.5 and is only available on Windows.  This PR attempts to update the Oculus manager to use 0.6 for windows, and 0.5 for other platforms.  

It also changes the 0.5 behavior to use a new window for output, which should always be correctly positioned, removing the need to fiddle with the main application's 'fullscreen state' or position.  Since the 0.6 SDK manages the Rift display window, this makes the different platforms less dissimilar in terms of code.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5faa53d7e9c1309599310cd11a042dd2b232e329 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6147844f937902f962091a3ca1a33adfe0bafff8 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 11bc838be9580528dd29be643d0cc5c83aecff4e available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5a1b7131603ae451ba15189cad093a96fba1a97e available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 43bddf23b4424c5d425945fc6bc66f1a68f188c1 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e9587d90c4c6902e92c335ff6d1980bd6098515f available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by jherico** [comment]
So the problem here is that oglplus requires either glew.h, gl3.h or glcorearb.h to build and none are available on Ubuntu by default.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5680d58a191c0a3e4a9455e4160133a931f9e543 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by sethalves** [comment]
On my (linux) system, I see:

mesa-common-dev:amd64: /usr/include/GL/glcorearb.h
libgles2-mesa-dev:amd64: /usr/include/GLES3/gl3.h

but no glew.h.

**Comment by jherico** [comment]
I don't know if the GLES3 version of gl3.h is sufficient, and I don't see glcorearb.h in mesa-common-dev on my Ubuntu 14 system.  On my system however, glew is available via libglew-dev.

Still, ideally I'd like to find a solution that doesn't require Leo to fiddle with the build system.

**Comment by jherico** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5680d58a191c0a3e4a9455e4160133a931f9e543 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 78c3ba0e2cf6005f2a7a5a455c292f6e40774954 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7b355fe69fdf5dc9a5659a2cbb27fe8e86a79a87 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e5a9fa8cdcabe8dc99cee9f9bf82988dfcf55fba available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by PhilipRosedale** [comment]
Where are we with this?   Ready to merge?

File: interface/src/ui/HMDToolsDialog.cpp [hunk 1]
```diff
@@ -244,13 +187,15 @@ void HMDToolsDialog::hideEvent(QHideEvent* event) {
...
+        // FIXME this is ineffective because it doesn't trigger the menu to 
+        // save the fact that VR Mode is not checked.
         leaveHDMMode();
     }
 }
 
 void HMDToolsDialog::screenCountChanged(int newCount) {
     if (!OculusManager::isConnected()) {
-        OculusManager::connect();
+        //OculusManager::connect();
```
Comment (sethalves) [inline interface/src/ui/HMDToolsDialog.cpp]: Is this how you intended to leave this?

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4f8ea3f624c78b1dba8a2d6659da2e19faf5d0be available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

File: interface/src/ui/HMDToolsDialog.cpp (hunk 1 above)
Comment (jherico) [inline interface/src/ui/HMDToolsDialog.cpp]: will fix.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2172445db76ce3c69727c495d5b930404220f512 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2172445db76ce3c69727c495d5b930404220f512 available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4633cabcecabc050817e7fc138e3ecea6d27db0b available at: https://gist.github.com/41009257612856fbbfdf<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5073.dmg

**Comment by ZappoMan** [comment]
tested on mac: passed

**Comment by ZappoMan** [comment]
basic test on windows

File: libraries/shared/CMakeLists.txt [hunk 2]
```diff
@@ -2,8 +2,20 @@ set(TARGET_NAME shared)
 
 # use setup_hifi_library macro to setup our project and link appropriate Qt modules
 # TODO: there isn't really a good reason to have Script linked here - let's get what is requiring it out (RegisteredMetaTypes.cpp)
-setup_hifi_library(Gui Network Script Widgets)
+setup_hifi_library(Gui Network OpenGL Script Widgets)
```
Comment (murillodigital) [inline libraries/shared/CMakeLists.txt]: this new library being linked to in shared is what's causing domain-server and assignment-client also depend on OpenGL - if that is by accident, this should be moved elsewhere

**Pull request merged**